enigma/src/main/java/cuchaz/enigma/analysis/index/InnerClassIndex.java

```diff
@@ -0,0 +1,54 @@
+package cuchaz.enigma.analysis.index;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.Multimap;
+
+import cuchaz.enigma.translation.representation.entry.ClassDefEntry;
+import cuchaz.enigma.translation.representation.entry.ClassEntry;
+
+public class InnerClassIndex implements JarIndexer {
+	private Multimap<ClassDefEntry, InnerClassData> innerClasses = ArrayListMultimap.create();
```

Comment:
This thing is wrong; we don't want to record one map entry for every `MethodHandles.Lookup` reference. To check if a class is an anonymous class, simply check its `EnclosingMethods` https://docs.oracle.com/javase/specs/jvms/se19/html/jvms-4.html#jvms-4.7.7 which asm somehow calls "OuterClass" and describes it incorrectly